scripts/manage/check_cluster_config.sh

```diff
@@ -14,89 +24,121 @@ if [ -z "$PROJECT_ID" ]; then
+    if [ "$CURRENT_CONTEXT_MATCH" ]; then
+      clear_current_context
+    fi
   fi
+}
 
-  # Now that we have retrieved credentials for each cluster in this project, query the configured contexts again.
-  CONTEXT_LIST=$(kubectl config get-contexts -o name)
-  CONTEXT_COUNT=$(echo "$CONTEXT_LIST" | sed '/^$/d' | wc -l)
+clear_current_context() {
+  sed -i"" -e"s/^current-context:.*$/current-context:/" ~/.kube/config
+}
 
+check_all_contexts() {
   if [ "$CONTEXT_COUNT" == "1" ]; then
-    bold "There is exactly one Kubernetes context configured: $CONTEXT_LIST"
-
-    # Since there is exactly one context configured, we'll set that as the current context.
+    # Since there is exactly one context configured, we'll set that as the current context (if a deployment is found).
     check_for_spinnaker_deployment $CONTEXT_LIST "" "select_context"
   else
-    bold "There are multiple Kubernetes contexts configured."
-
     CONTEXT_LIST=($CONTEXT_LIST)
 
     # Since there are multiple contexts configured, we will just query for Spinnaker deployments and generate
     # commands that can be used to select a current context.
     for c in "${CONTEXT_LIST[@]}"; do
       check_for_spinnaker_deployment $c "generate_command"
     done
+  fi
+}
+
+query_configured_contexts() {
+  CONTEXT_LIST=$(kubectl config get-contexts -o name)
+  CONTEXT_COUNT=$(echo "$CONTEXT_LIST" | sed '/^$/d' | wc -l)
+}
+
+get_all_project_cluster_credentials() {
+  RETRIEVED_PROJECT_CLUSTER_CREDENTIALS=true
+  bold "Querying for GKE clusters in project $PROJECT_ID..."
+
+  CLUSTER_LIST=$(gcloud beta container clusters list --format json --project $PROJECT_ID)
+  CLUSTER_COUNT=$(echo $CLUSTER_LIST | jq '. | length')
 
+  if [ "$CLUSTER_COUNT" == "0" ]; then
+    bold "No GKE clusters were found in project $PROJECT_ID."
+    exit 1
   fi
-else
-  bold "Using context $CURRENT_CONTEXT."
 
-  # There is a current context, so we will just query for a Spinnaker deployment.
-  check_for_spinnaker_deployment $CURRENT_CONTEXT
+  for (( i=0; i<$CLUSTER_COUNT; i++ )); do
+    # TODO: Determine implications of encountering non-zonal cluster here.
+    bold "Retrieving credentials from project $PROJECT_ID for cluster" \
+      "$(echo $CLUSTER_LIST | jq -r ".[$i].name")" \
+      "in zone $(echo $CLUSTER_LIST | jq -r ".[$i].zone")..."
+
+    gcloud container clusters get-credentials $(echo $CLUSTER_LIST | jq -r ".[$i].name") \
+      --zone $(echo $CLUSTER_LIST | jq -r ".[$i].zone") --project $PROJECT_ID
+  done
+
+  # Removing current context since gcloud container clusters get-credentials implicitly sets it.
+  # We want to avoid this behavior since there may be multiple clusters in this project.
+  # If there is exactly one cluster, the context will be automatically selected in the next steps anyway.
+  clear_current_context
+}
+
+bold "Querying for current Kubernetes context..."
+
+CURRENT_CONTEXT=$(kubectl config current-context)
+
+query_configured_contexts
```

Comment:
I might be wrong, but couldn't this be simplified to
```
query_configured_contexts

if [ "$CONTEXT_COUNT" != "0" ]; then
  check_all_contexts
fi

if [ -z "FOUND_MATCH_IN_PROJECT" -a -z "$RETRIEVED_PROJECT_CLUSTER_CREDENTIALS" ]; then
  get_all_project_cluster_credentials
  query_configured_contexts
  check_all_contexts
fi
```

Reply:
Actually, you could remove the `RETRIEVED_PROJECT_CUSTER_CREDENTIALS` variable too, I think.

Reply:
Thanks, that's much nicer.